src/driver/sap/SapQueryRunner.ts

```diff
@@ -202,15 +202,24 @@ export class SapQueryRunner extends BaseQueryRunner implements QueryRunner {
             const queryStartTime = +new Date()
             const isInsertQuery = query.substr(0, 11) === "INSERT INTO"
 
-            statement = databaseConnection.prepare(query)
+            if (parameters?.some(Array.isArray)) {
+                statement = await promisify(
+                    databaseConnection.prepare.bind(databaseConnection),
+                )(query)
+            }
 
-            const raw = await new Promise<any>((ok, fail) => {
-                statement.exec(parameters, (err: any, raw: any) =>
-                    err
-                        ? fail(new QueryFailedError(query, parameters, err))
-                        : ok(raw),
-                )
-            })
+            let raw: any
+            try {
+                raw = statement
```

Comment:
This condition is identical to the single-branch if statement above, you could merge the two.

Reply:
there is a `prepare.bind` not an exec as I see it. But I do not know how they differ in execution. Therefore, if it is only a preparation for the exec and that only has to happen if there is some parameter in the parameters array this looks reasonable.